irohad/main/CMakeLists.txt

```
@@ -69,3 +69,33 @@ target_link_libraries(irohad
     )
 
 add_install_step_for_bin(irohad)
+
+set(IROHA_ORIGIN "")
+set(IROHA_BRANCH "")
+set(IROHA_COMMIT_HASH "")
+
+execute_process(
+        COMMAND git config --get remote.origin.url
+        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+        OUTPUT_VARIABLE IROHA_ORIGIN
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+execute_process(
+        COMMAND git rev-parse --abbrev-ref HEAD
+        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+        OUTPUT_VARIABLE IROHA_BRANCH
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+execute_process(
+        COMMAND git log -1 --format=%h
```

Better use
```suggestion
        COMMAND git log -1 --format=%H
```
instead for long commit hash